tests/src/baseservices/threading/commitstackonlyasneeded/StackCommitCommon.cs

```diff
@@ -236,9 +236,18 @@ public class Finalizer
 
         public static void Run(Action action)
         {
-            new Finalizer(action);
+            //We need to allocate the object inside of a seperate method to ensure that
+            //the reference will be eliminated before GC.Collect is called. Technically
+            //even across methods we probably don't make any formal guarantees but this
+            //is sufficient for current runtime implementations.
+            CreateUnreferencedObject(action);
             GC.Collect();
             GC.WaitForPendingFinalizers();
         }
+
+        private static void CreateUnreferencedObject(Action action)
```

Comment:
You may wan to throw no-inline here for a good measure as well.

Reply:
thanks, on it